src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotReturnArrays.cs

```diff
@@ -26,14 +27,28 @@ public abstract class PropertiesShouldNotReturnArraysAnalyzer : DiagnosticAnalyz
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/0fss9skc.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Property);
+
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var symbol = (IPropertySymbol)context.Symbol;
+            if (symbol.Type.TypeKind == TypeKind.Array && !symbol.IsOverride)
+            {
+                if (symbol.GetResultantVisibility() == SymbolVisibility.Public && symbol.ContainingType.BaseType.MetadataName != "Attribute")
```

Comment:
Weren't you going to call symbol.IsAttribute() here? This check is a little looser and simply finds any type named Attribute.